src/homepage/howTo/custom-listeners.html

```diff
@@ -0,0 +1,237 @@
+                </p>
+
+                <header class="major">
+                    <h3><b>Starting James with BigMessageListener</b></h3>
+                </header>
+
+                <p>
+                    When you write a custom Listener, you have to compile it and place the compiled jar package inside "extensions-jars" directory
+                    to make James load your Listener when it starts up. But BigMessageListener is already available in James products.
+                    So we can use it directly without that extra step.
+                </p>
+
+                <p>We will take James cassandra product for the example</p>
+
+                <p>First, get template cassandra product configuration:</p>
+                <pre><code>
+$ git clone https://github.com/apache/james-project
+$ cp -rf james-project/dockerfiles/run/guice/cassandra/destination/conf conf
+                </code></pre>
+
+                <p>Then create the keystore file in the conf/ directory with the default password <code>james72laBalle</code>
+                <pre><code>
+$ keytool -genkey -alias james -keyalg RSA -keystore conf/keystore
+                </code></pre>
+
+                <p>
+                    Second, modify listener.xml configuration file in conf/ directory to use only BigMessageListener
+                    by specifying its full class name. We only need to use this Listener in our example.
+                </p>
+                <pre><code>
+&lt;listeners&gt;
+    &lt;listener&gt;
+        &lt;class&gt;org.apache.james.examples.custom.listeners.BigMessageListener&lt;/class&gt;
+    &lt;/listener&gt;
+&lt;/listeners&gt;
+                </code></pre>
+
+                <p>Finally, starting a James Server by docker compose</p>
+                <p>Getting James docker-compose.yml</p>
+                <pre><code>
+$ wget https://raw.githubusercontent.com/apache/james-project/master/dockerfiles/run/docker-compose.yml
+                </code></pre>
+                <p>Add the following volumes for james service:</p>
+                <pre><code>
+volumes:
+  - $PWD/conf:/root/conf/
+                </code></pre>
+
+                <p>
+                    When you are using your listeners, place the jar containing your listeners into "extensions-jars" directory
+                    and add a volume for it.
+                </p>
+                <pre><code>
+volumes:
+  - $PWD/extensions-jars:/root/extensions-jars/
+                </code></pre>
+                <p>
+                    In this case, you should compile James project and put its jar of module <b>examples/custom-listeners</b> to "extensions-jars".
+                    The jar name can be sightly different by the time because its name carries James version, but you can recognize it easily.
+                </p>
+                <pre><code>
+$ mkdir extensions-jars
+$ cd james-project
+$ mvn clean install -DskipTests -T 1C
```

Comment:
Can we write a command line more user friendly, were we compile only the needed project (custom-listeners) and download the dependencies?

Again compiling the FULL James source code proved to be a pretty high entry barrier...